zcash_primitives/src/transaction/components/orchard.rs

```diff
@@ -432,54 +423,50 @@ pub fn write_action_without_auth<W: Write, D: OrchardDomainCommon>(
 
 #[cfg(any(test, feature = "test-dependencies"))]
 pub mod testing {
+    use orchard::Bundle;
     use proptest::prelude::*;
 
     use crate::transaction::components::amount::testing::arb_amount;
-    use crate::transaction::components::Amount;
-    use crate::transaction::TxVersion;
+    use crate::transaction::{OrchardBundle, TxVersion};
     use orchard::bundle::{
         testing::{self as t_orch},
-        Authorized, Bundle,
+        Authorized,
     };
-    use orchard::orchard_flavor::{OrchardVanilla, OrchardZSA};
+    use orchard::orchard_flavor::OrchardZSA;
 
     prop_compose! {
         pub fn arb_bundle(n_actions: usize)(
             orchard_value_balance in arb_amount(),
             bundle in t_orch::BundleArb::arb_bundle(n_actions)
-        ) -> Bundle<Authorized, Amount, OrchardVanilla> {
+        ) -> OrchardBundle<Authorized> {
             // overwrite the value balance, as we can't guarantee that the
             // value doesn't exceed the MAX_MONEY bounds.
-            bundle.try_map_value_balance::<_, (), _>(|_| Ok(orchard_value_balance)).unwrap()
-        }
-    }
-
-    pub fn arb_bundle_for_version(
-        v: TxVersion,
-    ) -> impl Strategy<Value = Option<Bundle<Authorized, Amount, OrchardVanilla>>> {
-        if v.has_orchard() {
-            Strategy::boxed((1usize..100).prop_flat_map(|n| prop::option::of(arb_bundle(n))))
-        } else {
-            Strategy::boxed(Just(None))
+            OrchardBundle::OrchardVanilla(Box::new(bundle.try_map_value_balance::<_, (), _>(|_| Ok(orchard_value_balance)).unwrap()))
         }
     }
 
     prop_compose! {
+        #[allow(unreachable_code)]
         pub fn arb_zsa_bundle(n_actions: usize)(
             orchard_value_balance in arb_amount(),
             bundle in t_orch::BundleArb::arb_bundle(n_actions)
-        ) -> Bundle<Authorized, Amount, OrchardZSA> {
+        ) -> OrchardBundle<Authorized> {
             // overwrite the value balance, as we can't guarantee that the
             // value doesn't exceed the MAX_MONEY bounds.
-            bundle.try_map_value_balance::<_, (), _>(|_| Ok(orchard_value_balance)).unwrap()
+            let _bundle: Bundle<_, _, OrchardZSA> = bundle.try_map_value_balance::<_, (), _>(|_| Ok(orchard_value_balance)).unwrap();
```

Comment:
`: Bundle<_, _, OrchardZSA>` is redundent, so is the `_bundle` var

Reply:
fixed